src/ui-test/tests/camelk_java_debug_test.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+import {
+	EditorView,
+	VSBrowser,
+} from 'vscode-extension-tester';
+import { createIntegration, getIntegration, modifyCurrentFileToBeInvalid, prepareEmptyTestFolder, startIntegration } from '../utils/utils';
+import { cleanOutputView, sidebarIntegrationRemove } from '../utils/waitConditions';
+import * as uiTestConstants from '../utils/uiTestConstants';
+import { promiseFail, promiseSucceed } from '../utils/promiseUtils';
+
+const START_DEBUG_LABEL = uiTestConstants.startDebug;
+
+export function camelKJavaDebugTest() {
+
+	describe('Test Debug on Camel K Integrations from Side Bar', function () {
+
+		describe('Java Debug', function () {
+
+			const INTEGRATION_LABEL = 'java-debug-test';
+			const INTEGRATION_FILE = 'JavaDebugTest';
+
+			before(async function () {
+				this.timeout(uiTestConstants.TIMEOUT_60_SECONDS);
+				await createIntegration(INTEGRATION_FILE, 'Java', 'java');
+				await startIntegration(INTEGRATION_LABEL, 'Basic');
+			})
+
+			it('Check Java Debug available', async function () {
+				this.timeout(uiTestConstants.TIMEOUT_30_SECONDS);
+				const item = await getIntegration(INTEGRATION_LABEL);
+				const menu = await item.openContextMenu();
+				await promiseSucceed(VSBrowser.instance.driver.wait(() => menu.hasItem(START_DEBUG_LABEL), uiTestConstants.TIMEOUT_5_SECONDS));
```

Comment:
Why not directly:

```
await VSBrowser.instance.driver.wait(() => menu.hasItem(START_DEBUG_LABEL), uiTestConstants.TIMEOUT_5_SECONDS)
```

Reply:
First because that's not an assertion but counting on the code to work or fail whenever there is an exception.

Second because there are cases where promises could succeed or fail without having properly asserted the condition. We want to know if the wait command successfully completed and not just go forward.

There was more logic to it but I have to check the discussion because this was prompted by someone that requested it.